### ISSUE TITLE
Document Bytecompiler

### ISSUE THREAD

**Issue by jasonwilliams** [boa-dev/boa#1608]
There's no documentation for the [bytecompiler](https://github.com/boa-dev/boa/tree/82c578b434e99a73ae95e28c968990ce20e64843/core/engine/src/bytecompiler) we should add some

**Comment by postmeback** [comment]
The linked page is no longer existing. Is it good to close?

**Comment by jedel1043** [comment]
That's just because we changed the location of the bytecompiler. This is still unresolved.

**Comment by postmeback** [comment]
Can you paste the updated link? I want to document it.

**Comment by jedel1043** [comment]
I've updated the link from the issue :)

**Comment by jkrishnad** [comment]
Hey @jedel1043 is this issue still unresolved? and the page linked in here is no longer exist

**Comment by jedel1043** [comment]
@JkrishnaD Yep still unresolved. I changed the link to a permalink. Do you wanna pick up this?

**Comment by jkrishnad** [comment]
@jedel1043 i'll try this, so is there any reference for the docs like how it should look like.

**Comment by jedel1043** [comment]
@JkrishnaD That's the fun part, we don't know 🤣

This is kind of a big task because you would have to deeply analyze the code in there to come up with useful comments explaining what each part of the bytecompiler does. 
My suggestion is to first open a PR with some module documentation on the root module file explaining the overall architecture of the VM. After that, we can discuss how to structure the rest of the documentation.

**Comment by jkrishnad** [comment]
Now this is interesting 😅. So, I’ll start by adding module-level documentation explaining the VM architecture and bytecompiler flow.

**Comment by jkrishnad** [comment]
Hey @jedel1043, i have gone through the bytecompiler and i observed that most of the functions, strcuts and few implementations has already have the right documentation and i think line to line documentation doesn't make any sense.

Now I have planned to add some missing documentation for the functions and impl's and also thought to add a md file in docs folder to explain the architecture and all if need.

what do you think about this?
